intelmq/bots/collectors/tcp/collector.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+import socket
+import struct
+import sys
+
+from intelmq.lib.bot import CollectorBot
+
+
+class TCPCollectorBot(CollectorBot):
+
+    def init(self):
+        self.address = (self.parameters.ip, int(self.parameters.port))
+        self.connect()
+
+    def recvall(self, conn, n):
+        # Helper function to recv n bytes or return None if EOF is hit
+        data = b''
+        while len(data) < n:
+            packet = conn.recv(n - len(data))
+            if not packet:
+                return None
+            data += packet
+        return data
+
+    def process(self):
+        conn = None
+        try:
+            conn, addr = self.con.accept()
+            self.logger.info('Connection address: %s.', addr)
+            while True:
+                # Read message length and unpack it into an integer
+                raw_msglen = self.recvall(conn, 4)
+                if not raw_msglen:
+                    conn.send(b"OK")
+                    return
+
+                # Read the message data
+                msg = self.recvall(conn, struct.unpack('>I', raw_msglen)[0])
+                report = self.new_report()
+                report.add("raw", msg)
+                self.send_message(report)
+        except socket.error:
+            self.logger.exception("Reconnecting.")
+            self.con.close()
+            self.connect()
+        except AttributeError:
```

Comment:
What is that for? Only hides error message in my opinion.

Reply:
Agree, I did some test cases and in this state of code, I find this superfluous. I added a warning "Incomplete message received" instead.